### ISSUE TITLE
Naive auto generated Lua bindings crashes on ESP32

### ISSUE THREAD

**Issue by surfskidude** [swig/swig#2181]
I am generated Lua bindings and I noticed the following construction in the auto generated code:

    resultptr = (esp_err_t *) malloc(sizeof(esp_err_t));
    memmove(resultptr, &result, sizeof(esp_err_t));

I am running this on a microcontroller with limited memory and after running for a while, malloc returns NULL. Maybe the code should check the return from malloc?

Swig file:
https://github.com/RealTimeLogic/LspAppMgr-ESP32/blob/master/tools/esp.i

I am using SWIG-4.0.2

**Comment by ojwb** [comment]
Lack of handling for memory allocation failing in the generated wrappers is sadly a widespread and long-standing issue - e.g. see #277 and #1755.

I think one reason these haven't been addressed is that it's hard to handle well at the level it happens (and if the attempt to handle it needs to allocate memory that may itself fail).  However code like the above invokes undefined behaviour which could result in anything happening, not simply a guaranteed segfault.

**Comment by surfskidude** [comment]
After some more testing, it appears that the auto generated bindings leak memory. I have noticed that the memory usage keeps going up when I call the binding for a specific function from Lua. I do not have a debugger on the device, but I have looked at the generated code and I do not see any code releasing the memory allocated by the auto generated wrapper function. It's a bit hard to follow the auto generated code and it appears to set a Lua metatable on a Lua object, which includes a pointer to the "malloc" allocated memory. However, I do not see a GC method or anything else releasing this memory.

The function I have repeatedly tested calling from Lua is: mcpwm_set_duty_in_us
The wrapper function  _wrap_mcpwm_set_duty_in_us, which allocates the memory can be found on line 5863 in the following file: https://github.com/RealTimeLogic/LspAppMgr-ESP32/blob/master/main/esp_wrap.c. The memory not being released is in this function at 5884.

**Comment by ojwb** [comment]
I'm going to close as a duplicate of #277.
